### ISSUE TITLE
Vulkan: Reading from instance uniform produces invalid shader

### ISSUE THREAD

**Issue by WinstonHartnett** [godotengine/godot#48748]
**Godot version:**

`v4.0.dev.mono.custom_build.9b637faf6`

**OS/device including version:**

- Arch/Manjaro
- Radeon RX 570
- Mesa 21.0.3
- Vulkan

**Issue description:**

Reading from instance uniforms in shader code produces this error and the shader is no longer applied (i.e. default, white mesh):
```
servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp:162 - Condition "!shader_singleton->shader.version_is_valid(version)" is true.
```

In console:
```
...
[ ~3400 lines of generated shader]
ERROR: Error compiling Fragment shader, variant #9 (
#define USE_LIGHTMAP
).
   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:249)
ERROR: Failed parse:
ERROR: 0:2061: 'instance_uniforms_ofs' : no such field in structure
ERROR: 0:2061: '' : compilation terminated
ERROR: 2 compilation errors.  No code generated.

   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:250)
[ all ~3400 lines of generated shader ]
   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:253)                               
ERROR: Condition "!shader_singleton->shader.version_is_valid(version)" is true.                         
   at: set_code (servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp:162)
...
```

This indidual error repeats multiple times. [Here is the generated shader code](https://paste.ee/d/DiaAK).

**Steps to reproduce:**
1. Get [InstanceRepro.zip](https://github.com/godotengine/godot/files/6488370/InstanceRepro.zip)
2. Open Node3d.tscn, select `Node3d`
3. Navigate to Mesh > Material > Shader
4. Note that shader works while LOC that access instance uniforms are commented
5. Uncomment instance uniforms and save
6. Shader error

**Minimal reproduction project:**

[InstanceRepro.zip](https://github.com/godotengine/godot/files/6488370/InstanceRepro.zip)

**Comment by Calinou** [comment]
cc @Chaosus

**Comment by akien-mga** [comment]
Fixed by #48910.
